File: python/findsmallestlettergreaterthantarget.py
from typing import List


class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        """
        TC: O(n), SP: O(1)
        """
        i = 0
        n = len(letters)
        while i < n and ord(letters[i]) <= ord(target):
            i += 1
        return letters[i] if i != n else letters[0]


sol = Solution()
print(sol.nextGreatestLetter(["c", "f", "j"], "a"))

File: c++/findsmallestlettergreaterthantarget.cpp
class Solution {
public:
  char nextGreatestLetter(vector<char> &letters, char target) {
    int i = 0;
    int n = letters.size();
    while (i < n && letters[i] <= target) {
      i += 1;
    }
    if (i == n) {
      return letters[0];
    }
    return letters[i];
  }
};
